From 1c7e045d83ef2f945d9b226bdc1bdc6e900ea801 Mon Sep 17 00:00:00 2001
From: Martin Larsson <larssonmartin1998@gmail.com>
Date: Sat, 22 Feb 2025 11:49:09 +0000
Subject: [PATCH] [PATCH] type(lsp): Fix so vim.lsp.LocationOpts.reuse_win also
 focuses  the reused     window.

    Problem:
    When using the { reuse_win = true, }, the reused window is not focused.

    Solution:
    use vim.api.nvim_set_current_win to change the focus to the window that
    we're reusing if it only has a single result. This is useful when paired
    with diagnostic or declaration.
---
 runtime/lua/vim/lsp/buf.lua | 1 +
 1 file changed, 1 insertion(+)

diff --git a/runtime/lua/vim/lsp/buf.lua b/runtime/lua/vim/lsp/buf.lua
index c57fdbee18..13731d344d 100644
--- a/runtime/lua/vim/lsp/buf.lua
+++ b/runtime/lua/vim/lsp/buf.lua
@@ -215,6 +215,7 @@ local function get_locations(method, opts)
         local w = opts.reuse_win and vim.fn.win_findbuf(b)[1] or win
         api.nvim_win_set_buf(w, b)
         api.nvim_win_set_cursor(w, { item.lnum, item.col - 1 })
+        api.nvim_set_current_win(w)
         vim._with({ win = w }, function()
           -- Open folds under the cursor
           vim.cmd('normal! zv')
-- 
2.47.2

